adapters/repos/db/vector/hnsw/startup.go

```diff
@@ -88,10 +91,34 @@ func (h *hnsw) restoreFromDisk() error {
 			h.metrics.TrackStartupReadCommitlogDiskIO)
 		fdBuf := bufio.NewReaderSize(metered, 256*1024)
 
-		var valid int
-		state, valid, err = NewDeserializer(h.logger).Do(fdBuf, state, false)
+		var reader rwhasher.ReaderHasher
+		var checksumReader commitlog.ChecksumReader
+
+		checksumFileName := commitLogChecksumFileName(fileName)
+		hasChecksumFile, err := fileExists(checksumFileName)
 		if err != nil {
-			if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
+			return errors.Wrapf(err, "verifying checksum file existence %q", checksumFileName)
```

Comment:
This logic is repeated in the memory condensor as well, should we move it inside the `NewDeserializer` function?
That would also simplify the parameters of most of `Deserializer` methods

Reply:
managed to de-duplicate condensor code by reusing commit loggers but this initialization steps may even be more general than this steps of loading the checksum file. I got your point and will take a second look at it